File: agents/java-agent/src/main/java/com/hmdev/messaging/agent/api/ConnectionChannelApiFactory.java
package com.hmdev.messaging.agent.api;



import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import com.hmdev.messaging.agent.api.impl.MessagingChannelApi;



public abstract class ConnectionChannelApiFactory {
    private static final Logger logger = LoggerFactory.getLogger(ConnectionChannelApiFactory.class);

	private ConnectionChannelApiFactory(){
		super();
	}
	
	/**
	 * Create a ConnectionChannelApi for the given descriptor. The optional developerApiKey
	 * will be applied to the created API instance (X-Api-Key header) if provided.
	 */
	public static ConnectionChannelApi getConnectionApi(String descriptor) {
		return getConnectionApi(descriptor, null);
	}

	public static ConnectionChannelApi getConnectionApi(String descriptor, String developerApiKey) {
		if(descriptor.matches("^https?://.*")){
			return new MessagingChannelApi(descriptor, developerApiKey);
		}else{
			throw new RuntimeException("Connection channel descriptor is not supported");
		}
	}
}

File: agents/python-agent/hmdev/messaging/agent/api/connection_channel_api_factory.py
import re
from typing import Optional
from hmdev.messaging.agent.api.impl.messaging_channel_api import MessagingChannelApi


class ConnectionChannelApiFactory:
    @staticmethod
    def get_connection_api(remote_url: str, developer_api_key: Optional[str] = None) -> MessagingChannelApi:
        if re.match(r"^https?://", remote_url):
            return MessagingChannelApi(remote_url, developer_api_key=developer_api_key)
        else:
            raise RuntimeError("Connection channel descriptor is not supported")
